corehq/apps/app_manager/models.py

```diff
@@ -4078,8 +4094,15 @@ class ApplicationBase(VersionedDoc, SnapshotMixin,
     # always false for RemoteApp
     case_sharing = BooleanProperty(default=False)
 
+    build_profiles = SchemaDictProperty(BuildProfile)
+
+    # each language is a key and the value is a list of multimedia referenced in that language
+    media_language_map = SchemaDictProperty(MediaList)
```

Comment:
All menu media is sent to down with all profiles. At the time there was not a good way to determine which media was used for which languages on the menus without parsing build xml, which was determined to be out of scope. After discussing requirements with the icds team, that compromise was seen as fine.
It is definitely possible I missed a good way to find that or something has changed that would now make that easier.